### PR TITLE
FAQ: what is a package

### PR DESCRIPTION
**Pull request by ecki**
Clarify that package talks about deployment units.
Introduce the idea that it can apply to other constructs as well.
Add Java as example technology since it has different terminology.

This fixes semver#189

### CODE REVIEW

**Comment by ecki** [comment]
Should I raise the version of the document to 2.0.1 in this PR or keep that seperate?

**Comment by FichteFoll** [comment]
It doesn't touch the actual spec, so no imo.

Regardless, bumping the version should only occur when a release is made by updating semver.org since you wouldn't know what kind of changes would eventually end up in that release until it is frozen.

File: semver.md [hunk 1]
```diff
@@ -319,6 +319,22 @@ Abbreviating "version" as "v" is often seen with version control. Example:
 `git tag v1.2.3 -m "Release version 1.2.3"`, in which case "v1.2.3" is a tag
 name and the semantic version is "1.2.3".
 
+### What does SemVer consider a package?
+
+The meaning of package in this specification describes a deployment unit.
+It could be built from a single or multiple source artifacts. But it can
```
Comment (jwdonahue) [inline semver.md]: > ... artifacts. But it can...

Should either be "...artifacts, but it can..." or "...artifacts. It can..."

File: semver.md [hunk 2]
```diff
@@ -319,6 +319,22 @@ Abbreviating "version" as "v" is often seen with version control. Example:
...
+also be an assembly of multiple binary artifacts.
+
```
Comment (jwdonahue) [inline semver.md]: No need for one sentence paragraphs here.  Just connect the above to the one below and then the "however" in that lower sentence will not be cognitively burdensome.

File: semver.md [hunk 3]
```diff
@@ -319,6 +319,22 @@ Abbreviating "version" as "v" is often seen with version control. Example:
...
+Semantic versioning concepts can however also be applied to logical
+collections like an interface, an API or any other concept which
+undergoes changes and can be assigned a version.
+
+In Java for example a typical deployment unit would be a JAR file and a
+Java package is typically not versioned independenly.
+However with modular technologies (like *OSGi bundles*) the Java package
```
Comment (jwdonahue) [inline semver.md]: Very minor nit: this sentence should start on the line above.

**Comment by jwdonahue** [comment]
I started a review with some comments.  Thank you @ecki for contributing this, I hope it is accepted.  When it is, please don't forget to close #189.

File: semver.md (hunk 3 above)
Comment (ecki) [reply semver.md]: Do you mean breaking the line after ‚However with...‘? 

(i can do that, I am just used to breaking lines by sentence for markup, which makes generally more stable diffs. But I will try to fit in the existing style)

File: semver.md (hunk 2 above)
Comment (ecki) [reply semver.md]: Yes correct, was doing too much Asciidoc lately. And I will remove ‚the meaning‘

File: semver.md (hunk 3 above)
Comment (ecki) [reply semver.md]: Correction: will line join it as well

Comment (jwdonahue) [reply semver.md]: > "which makes generally more stable diffs"

Oh, I hadn't thought of that. Like I said, it was a nit, perhaps you should ignore me on this one.

File: semver.md (hunk 2 above)
Comment (jwdonahue) [reply semver.md]: "..." means whatever came before or whatever comes after. No need to insert them in the text.

File: semver.md (hunk 3 above)
Comment (FichteFoll) [reply semver.md]: Relevant: http://rhodesmill.org/brandon/2012/one-sentence-per-line/

**Comment by FichteFoll** [comment]
It will be closed automatically because the phrase "fixes #189" is included in the OP.

**Comment by ecki** [comment]
I have updated the Pull Request. I use the already existing formatting (i.e. not use line-per-sentence format). Please let me know what you think now.

File: semver.md [hunk 4]
```diff
@@ -319,6 +319,24 @@ Abbreviating "version" as "v" is often seen with version control. Example:
 `git tag v1.2.3 -m "Release version 1.2.3"`, in which case "v1.2.3" is a tag
 name and the semantic version is "1.2.3".
 
+### What does SemVer consider a package?
+
+With "package" SemVer describes deployment units, independent of the actual
+technology used. Packages could be built from source artifacts or assemble
+binary artifacts.
+
+An actual eco-system might have a different terminology. For example In *Java*
```
Comment (jwdonahue) [inline semver.md]: [Ecosystem](http://www.dictionary.com/browse/ecosystem?s=t) is one word, no need for the hyphen.

Comment (ecki) [reply semver.md]: Thanks @jwdonahue  - pushed a new version.

Comment (jwdonahue) [reply semver.md]: Every morning I seem to see something new. Let's accumulate some comments for a few days and see where it lands.  I think we're almost there.  The second paragraph looks fine to me, but I nearly dislocated a cog on the last sentence in the first paragraph. Either the paragraph needs another sentence or ... I don't know yet.  I need to let it ferment.

**Comment by sander1095** [comment]
This has been dead for a while, commenting to revive :)

**Pull request closed without merging**